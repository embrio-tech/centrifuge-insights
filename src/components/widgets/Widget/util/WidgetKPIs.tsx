import { Tooltip } from 'antd'
import React from 'react'
// import './WidgetKPIs.less'

export interface WidgetKPI {
  label: string
  value?: number | string
  unit?: string
}

interface WidgetKPIsProps {
  kpis: WidgetKPI[]
  title?: string
  className?: string
}

export const WidgetKPIs: React.FC<WidgetKPIsProps> = (props) => {
  const { className, kpis, title } = props

  return (
    <div className={className}>
      <h4 className='text-sm'>{title || 'KPIs'}</h4>
      <ul className='text-xs mb-0'>
        {kpis.map(({ label, value, unit }, index) => (
          <li key={index} className='flex h-5'>
            <div className='grow shrink truncate font-light'>{label}</div>
            {value !== undefined && (
              <div className='pl-2'>
                <Tooltip placement='left' title={label}>
                  {value}
                </Tooltip>
              </div>
            )}
            {unit !== undefined && <div>{unit}</div>}
          </li>
        ))}
      </ul>
    </div>
  )
}
