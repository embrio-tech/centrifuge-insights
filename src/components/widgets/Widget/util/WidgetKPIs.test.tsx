import React from 'react'
import renderer from 'react-test-renderer'
import { WidgetKPIs, WidgetKPI } from './WidgetKPIs'

describe('WidgetKPIs should', () => {
  test('match snapshot!', () => {
    const kpis: WidgetKPI[] = [
      {
        label: 'Pool value growth',
        value: 58.43,
        suffix: '%',
      },
      {
        label: 'Liquidity reserve as % of pool value',
        value: 24.2,
        suffix: '%',
      },
      {
        label: '',
      },
      {
        label: '# of loans',
        value: 0,
      },
    ]

    const tree = renderer.create(<WidgetKPIs kpis={kpis} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

export default null
