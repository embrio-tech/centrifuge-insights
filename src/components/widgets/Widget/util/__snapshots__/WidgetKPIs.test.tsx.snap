// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetKPIs should match snapshot! 1`] = `
<div>
  <h4
    className="text-sm"
  >
    KPIs
  </h4>
  <ul
    className="text-xs mb-0"
  >
    <li
      className="flex h-5 mb-1"
    >
      <div
        className="grow shrink truncate font-light"
      >
        Pool value growth
      </div>
      <div
        className="pl-2"
      >
        <span
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          58.43
        </span>
      </div>
      <div>
        %
      </div>
    </li>
    <li
      className="flex h-5 mb-1"
    >
      <div
        className="grow shrink truncate font-light"
      >
        Liquidity reserve as % of pool value
      </div>
      <div
        className="pl-2"
      >
        <span
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          24.2
        </span>
      </div>
      <div>
        %
      </div>
    </li>
    <li
      className="flex h-5 mb-1"
    >
      <div
        className="grow shrink truncate font-light"
      />
    </li>
    <li
      className="flex h-5 mb-1"
    >
      <div
        className="grow shrink truncate font-light"
      >
        # of loans
      </div>
      <div
        className="pl-2"
      >
        <span
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          0
        </span>
      </div>
    </li>
  </ul>
</div>
`;
